tool.ts

```diff
@@ -249,46 +249,49 @@ export async function downloadTool(
             }
 
             tl.debug('creating stream');
-            let file: NodeJS.WritableStream = fs.createWriteStream(destPath);
-            file.on('open', async (fd) => {
-                try {
-                    let stream = response.message.pipe(file);
-                    stream.on('close', () => {
-                        tl.debug('download complete');
-                        let fileSizeInBytes: number;
-                        try {
-                            fileSizeInBytes = _getFileSizeOnDisk(destPath);
-                        }
-                        catch (err) {
-                            fileSizeInBytes = NaN;
-                            tl.warning(`Unable to check file size of ${destPath} due to error: ${err.Message}`);
-                        }
-
-                        if (!isNaN(fileSizeInBytes)) {
-                            tl.debug(`Downloaded file size: ${fileSizeInBytes} bytes`);
-                        } else {
-                            tl.debug(`File size on disk was not found`);
-                        }
-
-                        if (!isNaN(downloadedContentLength) &&
-                            !isNaN(fileSizeInBytes) &&
-                            fileSizeInBytes !== downloadedContentLength) {
-                            tl.warning(`Content-Length (${downloadedContentLength} bytes) did not match downloaded file size (${fileSizeInBytes} bytes).`);
-                        }
-
-                        resolve(destPath);
-                    });
-                }
-                catch (err) {
+            const file: NodeJS.WritableStream = fs.createWriteStream(destPath);
+            file
+                .on('open', async (fd) => {
+                    try {
+                        response.message
```

Comment:
Sorry for the late comment on this.  Couple comments:
  1. Consider putting this change behind a feature flag
  2. Consider adding telemetry so we can see how often the 'aborted' code path is hit in Kusto to understand the impact of this change

  (while we're in there, we might want to add telemetry to the other 'catch' blocks as well)

Reply:
Thank you for your review point!

@merlynomsft I think that FF is an amazing practice for the rollout of something risky, for cases when we just include additional try/catch to parts that previously were ignored it is pretty safe. Also, this is a common library that is used in tasks and might be re-used somewhere else, we should have strict behavior for each method, and we should avoid changing the way how it works on the fly.
Thanks for the second item: @sergey-koryshev  could you please check is this tl.warning available in the telemetry?

Reply:
[this discussion isn't blocking for the PR, feel free to resolve as needed]

> or cases when we just include additional try/catch to parts that previously were ignored it is pretty safe

I agree there is a risk/benefit analysis as implementing even a simple FF is pretty expensive especially due to the server-side churn (so we should use best judgement!)

> test these changes in in-the-box tasks using FF/Semantic version approach

Adding a buildconfig (semversion) has additional costs (e.g. increase in build times for tasks); an if-else FF is relatively cheap; so it's better to prefer the later when possible.

My thought for library changes:
1. For 1st party tasks or code consuming the library, functional changes are gated by feature flag (off by default, enabled using safe rollout)
   1. One possible way to implement would be to add an additional API that first party tasks call into to opt into FF behavior
1. For 3rd party tasks (or wherever else may consume the library), functional changes are on by default

Reply:
Hi @merlynomsft, as for telemetry, after the PR all `aborted` errors will be in Kusto and we can see how often this happens.